src/TapeChart.js

```diff
@@ -0,0 +1,48 @@
+class TapeChart {
+  constructor(users, rooms, bookings) {
+    this.users = users;
+    this.rooms = rooms;
+    this.bookings = bookings;
+  }
+
+getAvailableRoomsByDate(date) {
+    let availableByDate = this.rooms.reduce((acc, room) => {
+      let bookingsByRoom = this.bookings.filter(booking =>
+        booking.roomNumber === room.number);
+      if (!bookingsByRoom.some(booking => booking.date === date)) {
+        acc.push(room);
+      }
+        return acc;
+    }, [])
+  return availableByDate;
+}
+
+getOccupancy(date) {
+  return 100 - (this.getAvailableRoomsByDate(date).length
+    / this.rooms.length * 100);
+}
+
+getBookingsByDate(date) {
```

Comment:
I am going to use this to refactor, this is much cleaner then what I wrote.

Reply:
Nice!  I love hearing that.